pymc3_hmm/step_methods.py

```diff
@@ -140,13 +162,29 @@ def __init__(self, var, values=None, model=None):
 
         # We compile a function--from a Theano graph--that computes the
         # total log-likelihood values for each state in the sequence.
-        dependents_log_lik = model.fn(
-            tt.sum([v.logp_elemwiset for v in self.dependent_rvs], axis=0)
-        )
+        var_sample = model.test_point[var.name]
+
+        log_likelihood_values = []
+        for i, dependent_rv in enumerate(self.dependent_rvs):
+            number_of_state = len(dependent_rv.distribution.comp_dists)
+
+            log_p_t = []
+            for state_i in range(number_of_state):
+                logp_t = dependent_rv.logp_elemwiset
+                log_p_t.append(traverse_graph_and_replace(logp_t))
+            log_p_t = tt.stack(log_p_t)
+
+            log_likelihood_values.append(log_p_t)
+
+        temp = tt.sum(log_likelihood_values, axis = 0)
+
+        dependents_log_lik = model.fn(temp)
```

Comment:
We should see how well this change alone performs.  I think this could already be an improvement, even without the in-place change.